subsys/bluetooth/host/Kconfig

```
@@ -15,8 +15,8 @@ config BT_LONG_WQ_STACK_SIZE
 	# Hidden: Long workqueue stack size. Should be derived from system
 	# requirements.
 	int
-	default BT_HCI_ECC_STACK_SIZE if BT_TINYCRYPT_ECC
-	default 1024
+	default BT_HCI_ECC_STACK_SIZE if (BT_TINYCRYPT_ECC && (BT_HCI_ECC_STACK_SIZE > 1300))
+	default 1300
```

Did you consider something like the following? 

```suggestion
	default 1300 if BT_GATT_CACHING
	default 1024
```

In the end, it is going to be a bit tricky w.r.t multiple `defaults` for non-constant values (like `BT_HCI_ECC_STACK_SIZE`).

---

Since the long workqueue is used to process both ECC operations and GATT DB hash, we need to take greater stack size from the ones that are needed for both operations. Fact that BT_HCI_ECC_STACK_SIZE is configurable makes us unable to use simpler solution (user may increase this stack size and we would still use smaller value, because if was defined for case where GATT caching is enabled).

---

As long as BT_HCI_ECC_STACK_SIZE option is available I would prefer to ensure that it's properly applied here. At some point it may be a good idea to completely remove it (and inform users that which value should be increased instead).

---

hm that sounds to me like this should be defined with a macro or something that has more control

---

> Please keep in mind that user cannot directly assign BT_LONG_WQ_STACK_SIZE too.

Not directly, but they can easily modify `BT_HCI_ECC_STACK_SIZE` and thus indirectly modify `BT_LONG_WQ_STACK_SIZE`. 

My point was that if `BT_GATT_CACHING=n`, then there's no need to increase this stack size.

---

I would opt for removing the BT_HCI_ECC_STACK_SIZE, since there is no such stack anymore. We could explicitly inform that ECC operations are currently performed in the Bluetooth long workqueue.

---

Agreed. `BT_HCI_ECC_STACK_SIZE` is currently only used to determine the `BT_LONG_WQ_STACK_SIZE`.

If we remove that, we can end up with something like

```c
config BT_LONG_WQ_STACK_SIZE
	# Hidden: Long workqueue stack size. Should be derived from system
	# requirements.
	int
	default 1300 if BT_GATT_CACHING
	default 1140 if BT_TINYCRYPT_ECC
	default 1024
```

And all should be good :) 

The `default X if Y` is of course a bit dangerous if someone decides to `BT_LONG_WQ_STACK_SIZE=700` which would cause both of the above to break (available with `depends on` or `BUILD_ASSERT`s

---

I guess we can add a safeguard BUILD_ASSERT in long_wq.c that double-checks the kconfig requirements.

---

Is it possible to set a conditional range?

---

If we order defaults in descending order we should be fine.
I also added prompt to allow user to modify BT_LONG_WQ_STACK_SIZE if necessary (Kconfig helps explicitly informs that: `Should not be changed unless absolutely necessary`, so user should be aware what he's doing if he changes this value).

---

Changes are already applied. Thanks for the comments @Thalley :)

---

> Is it possible to set a conditional range?

I would stick with approach providing multiple defaults and allowing user to change it if he is aware what he's doing. User could e.g. tailor stack if used architecture works fine also with smaller stacks. He could also increase the stack size if necessary.